R/spatial_block_cv.R

```diff
@@ -170,3 +197,56 @@ random_block_cv <- function(data, grid_blocks, v) {
 # Check sparse geometry binary predicate for empty elements
 # See ?sf::sgbp for more information on the data structure
 sgbp_is_not_empty <- function(x) !identical(x, integer(0))
+
+filter_grid_blocks <- function(grid_blocks, data) {
+  block_contains_points <- purrr::map_lgl(
+    sf::st_intersects(grid_blocks, data),
+    sgbp_is_not_empty
+  )
+  if ("data.frame" %in% class(grid_blocks)) {
+    grid_blocks[block_contains_points, ]
+  } else {
+    grid_blocks[block_contains_points]
+  }
```

Comment:
Do [we need `drop = FALSE` here](https://adv-r.hadley.nz/subsetting.html#simplify-preserve)?